pkg/controllers/user/pipeline/remote/github/github.go

```diff
@@ -287,6 +289,27 @@ func (c *client) GetDefaultBranch(repoURL string, accessToken string) (string, e
 
 }
 
+func (c *client) GetHeadCommit(repoURL string, branch string, credential *v3.SourceCodeCredential) (string, error) {
+
+	if credential != nil {
+		userName := credential.Spec.LoginName
+		token := credential.Spec.AccessToken
+		repoURL = strings.Replace(repoURL, "://", "://"+userName+":"+token+"@", 1)
+	}
+
+	return getBranchHeadCommit(repoURL, branch)
+}
+
+func getBranchHeadCommit(url, branch string) (string, error) {
+	cmd := exec.Command("git", "ls-remote", url, branch)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", errors.Wrap(err, string(output))
+	}
+	strs := strings.Split(string(output), "\t")
+	return strs[0], nil
```

Comment:
feels like we need a check for strs len

Reply:
it comes from Split so it will have at least 1 in size anyway